curriculum/challenges/english/08-coding-interview-prep/project-euler/problem-35-circular-primes.english.md

```diff
@@ -63,37 +63,55 @@ circularPrimes(1000000);
 
 
 ```js
-const circularPrimes = (n) => {
-  const primeCheck = (num) => {
-    if (num === 1) {
-      return false;
-    }
-    for (let i = 2; i <= Math.floor(Math.sqrt(num)); i++) {
-      if (num % i === 0) {
-        return false;
-      }
+function rotate(n) {
+  if (String(n).length == 1) return n;
+  return "" + (n % 10) + Math.floor(n / 10);
+}
+function circularPrimes(n) {
+  const primes = [];
+  let count = 0;
+
+  // Making primes array
+  for (let i = 0; i < n; i++) {
```

Comment:
Which two loops? `isPrime` and `circularPrimes` loop? 
I'm not sure how, because `isPrime` is checking if the number is a prime number for the rotations too, so for example if we have this `193939`, then `circularPrimes` loops till `20000` but `isPrime` checks if `939391` is a prime which is > 20000.

Maybe I'm missing something. Can you show how they can be combined?

Reply:
```js
const primes = [0, 0, 2];
let bound = 10;
while (bound < n) { bound = bound * 10; }
for (let i = 3; i < bound; i += 2) {
    primes.push(i);
    primes.push(0);
}
```

Reply:
Ah, got it.

Reply:
if we calculate primes to `bound`we'll not need the `isPrime` function.

Reply:
How? `bound < n`, whereas `isPrime()` is checking for `num > n`.

Reply:
`boud` is nearest `10^k > n`

Reply:
So you want to make primes array of length `10^k`?

Reply:
Yes, because we need to check primes in this boundaries

Reply:
@ValeraS I'm not sure if thats the best idea, considering what you're suggesting will take x5 times more memory than the current solution. Example `n = 200000` then the array size will be `n = 1000000`.

Do we not care about space complexity? Or is it only time complexity we care about?

P.S. Sorry for late reply, I've been travelling.

Reply:
We have constraints `100< n < 1 000 000`, and for `n = 1 000 000` array size will be `1 000 000` anyway so, IMO, it's not a problem.

Reply:
The tests will create a new array for each individual tests, isnt it?

So for `n = 200000`, it'll make an array of length `1000000`, for `n = 5000000`, will make an array of length `n = 1000000`.

I'll play with your idea, and will test out if it improves on the current solution and update you by tonight.

Reply:
@ValeraS Can you check this [repl](https://repl.it/@ezioda004/AcademicAdventurousParallelcomputing)? I've implemented the bound one in the first one.
Sometimes the one with bound is pulling, other times the one with `isPrime()` is pulling ahead.

Shall I go ahead and make changes with the bound one or is there any additional changes should I do?

Reply:
@ezioda004 I think, it's ok then. But can you check this [repl](https://repl.it/@ValeraS1/CheerySphericalDrawing)

Reply:
I suppose since the input are small, `.slice()` is more optimized.
Fixed